contracts/zero-ex/contracts/src/features/MetaTransactionsFeature.sol

```diff
@@ -347,17 +348,34 @@ contract MetaTransactionsFeature is
                     state.mtx.value
                 ).rrevert();
         }
-        // Must be signed by signer.
-        try
-            ISignatureValidatorFeature(address(this))
-                .validateHashSignature(state.hash, state.mtx.signer, state.signature)
-        {}
-        catch (bytes memory err) {
+
+        if (state.signature.length != 66) {
+            LibSignatureRichErrors.SignatureValidationError(
+                LibSignatureRichErrors.SignatureValidationErrorCodes.INVALID_LENGTH,
+                state.hash,
+                state.mtx.signer,
+                state.signature
+            ).rrevert();
+        }
+
+        LibSignature.Signature memory sig = LibSignature.Signature({
+            signatureType: LibSignature.SignatureType(uint8(state.signature[65])),
+            v: uint8(state.signature[0]),
+            r: state.signature.readBytes32(1),
+            s: state.signature.readBytes32(33)
+        });
+
+        if (LibSignature.getSignerOfHash(state.hash, sig) != state.mtx.signer) {
             LibMetaTransactionsRichErrors
                 .MetaTransactionInvalidSignatureError(
                     state.hash,
                     state.signature,
-                    err
+                    LibSignatureRichErrors.SignatureValidationError(
```

Comment:
I think we can drop the nested error here and instead just use `LibSignatureRichErrors.SignatureValidationError` (@dorothy-zbornak opinion?). If we do want to keep the nested error however, we'd probably want the invalid length error above to also be nested.

Reply:
My rationale here was to keep the existing tests working. So the errors are (hopefully) the same as what we used to return.

I'd definitely be happy to drop the nested error; I'm just pointing out that this would theoretically be a breaking change.

Reply:
Oh, but I think it's true that the invalid length error _should_ be nested to preserve the earlier behavior.

So I failed at keeping the errors the same. Let's make a call on nesting or not, and I'll make one change or the other.

Reply:
I'm down for removing the nested error. Probably not terribly useful in production.